specification.md

```diff
@@ -133,122 +154,130 @@ Example meta file for an Entry:
-  array indicating the units of the associated field in the data.
-
-The following attribute is only required for datasets with a discrete timebase:
-
-- **sampling\_rate:** A nonzero number indicating the sampling rate of the data,
-  in samples per second (Hz).
-  May be any numerical datatype.
-
-The following attributes are optional:
-
-- **unit\_scale**: A multiplier to scale the raw data to match the **units**. Useful
-  when raw integer data must be converted to a floating point number to match the correct units.
-  If **units** is an array, **unit\_scale** must be an array of the same shape.
-- **offset:** Indicates the start time of the dataset relative to the start of
-  the entry, defined by the timebase of the dataset. For discrete timebases, the
-  units must be in samples; for continuous timebases, the units must be the same
-  as the units of the dataset. If this attribute is missing, the offset shall be
-  assumed to be zero.
-- **uuid:** A universally unique ID for the dataset (see
+- **`units`:** A string giving the units of the data. `units` must, with two 
+  exceptions, be an SI unit abbreviation. Inference of its meaning is case-
+  sensitive (e.g., `s` means seconds, but `S` means Siemens). The first
+  exception to this rule is the value `samples`, which is dimensionless in SI
+  notation. The second exception is a null value, to be used if the units are
+  unknown. The null value in YAML is `null`; in Python, use `None`. An empty
+  string is also treated as a `null` value. For complex event data, this
+  attribute must be an array. Event datasets **must** have units of either "s"
+  or "samples"; sampled datasets **must not** use those units.
+
+The following attributes are **required** for all sampled datasets:
+
+- **`sampling_rate`:** A nonzero positive number indicating the sampling rate
+  of the data, in samples per second (Hz). May be either an integer or a
+  floating-point value.
+- **`dtype`:** the numeric type of the data, such as 16-bit integer or 32-bit
+  float
+- **`n_channels`:** the number of channels in the dataset
+
+The following attribute is **required** for event datasets with `units` of
+"samples":
+
+- **`sampling_rate`:** as described above for sampled datasets.
+
+The following attributes are defined by the spec, but are optional:
+
+- **`unit_scale`:** A multiplier to scale the raw data to match the **units**.
+  Useful when raw integer data must be converted to a floating point number to
+  match the correct units. If `units` is an array, `unit_scale` must be an
+  array of the same shape.
+- **`offset`:** Indicates the start time of the dataset relative to the
+  timestamp of the entry. For discrete timebases, the units must be in samples;
+  for continuous timebases, the units must be the same as the units of the
+  dataset. If this attribute is missing, the offset is assumed to be zero. If
+  it is present, it must be accompanied by the `offset_units` attribute. Note
```

Comment:
The is a break from the arf spec. Is there are reason for this? Here's the Arf spec's description:

offset: Indicates the start time of the dataset relative to the start of the entry, defined by the timebase of the dataset. For discrete timebases, the units must be in samples; for continuous timebases, the units must be the same as the units of the dataset. If this attribute is missing, the offset shall be assumed to be zero.